src/documents.rs

```diff
@@ -387,6 +412,41 @@ mod tests {
         Ok(())
     }
 
+    #[meilisearch_test]
+    async fn test_delete_documents_with(client: Client, index: Index) -> Result<(), Error> {
+        setup_test_index(&client, &index).await?;
+        index
+            .set_filterable_attributes(["id"])
+            .await
+            .unwrap()
+            .wait_for_completion(&client, None, None)
+            .await
+            .unwrap();
+        let mut query = DocumentDeletionQuery::new(&index);
+        query.with_filter("id = 1");
+
+        index
+            .delete_documents_with(&query)
+            .await
+            .unwrap()
+            .wait_for_completion(&client, None, None)
+            .await
+            .unwrap();
+        let document_result = index.get_document::<MyObject>("1").await;
```

Comment:
I think we should try to parse all the error codes that can be returned by this route, I left a few comments on all the things we should add to the test.
Also, I think we can get rid of all the `.unwrap()`.

```suggestion
    async fn test_delete_documents_with(client: Client, index: Index) -> Result<(), Error> {
        setup_test_index(&client, &index).await?;
        
        //We should ensure it doesn't work if we don’t use any parameter in the builder and we can parse the new error code
        let mut query = DocumentDeletionQuery::new(&index);
        index.delete_documents_with(&query) // TODO
        // TODO assert thingy
        
        // Then, just to be sure, it would also be nice to ensure we can handle the error returned by meilisearch in case the field is not filterable
        query.with_filter("id = 1");
         index.delete_documents_with(&query) // TODO
        // TODO assert thingy
        
        index
            .set_filterable_attributes(["id"])
            .await
            .unwrap() // you can use `?` to reduce the size of the test I think 🤔 
            .wait_for_completion(&client, None, None)
            .await
            .unwrap(); // same
        let mut query = DocumentDeletionQuery::new(&index);
        
        index
            .delete_documents_with(&query)
            .await
            .unwrap() // same
            .wait_for_completion(&client, None, None)
            .await
            .unwrap(); // same
        let document_result = index.get_document::<MyObject>("1").await;
        
        // one last thing that we should test since it returns a new error code is that we get the right error code with a bad filter
        query.with_filter("doggo");
```